main/views/partials.py

```diff
@@ -4,29 +4,27 @@
 from django.contrib.auth.decorators import login_required
 from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render
-from django.utils.timezone import localtime
+from django_tables2 import RequestConfig
 
 from main.models import DruncMessage
+from main.tables import DruncMessageTable
 from main.views.utils import handle_errors
 
 
 @login_required
 @handle_errors
 def messages(request: HttpRequest, topic: str) -> HttpResponse:
-    """Search and render Kafka messages from the database."""
+    """View function to display messages for a given topic."""
     search = request.GET.get("search", "")
     records = DruncMessage.objects.filter(
         topic__regex=settings.KAFKA_TOPIC_REGEX[topic], message__icontains=search
-    )
+    ).order_by("-timestamp")
 
-    # Time is stored as UTC. localtime(t) converts this to our configured timezone.
-    messages = [
-        f"{localtime(record.timestamp).strftime('%Y-%m-%d %H:%M:%S')}: {record.message}"
-        for record in records
-    ]
+    table = DruncMessageTable(records)
+    RequestConfig(request).configure(table)
```

Comment:
This will by default paginate the table which is also not compatible with how we're using htmx here. I think you can do 
```suggestion
    table = DruncMessageTable(records)
    RequestConfig(request, paginate=False).configure(table)
```
to disable.